lib/finch/client/api/connection.rb

```diff
@@ -3,7 +3,14 @@
 module Finch
   module Client
     class API
-      class APIError < StandardError; end
+      class APIError < StandardError
+        attr_reader :response
+
+        def initialize(message, response)
+          super(message)
+          @response = response
```

Comment:
Do we have a good idea what the circumstance might be where this would be returned on the frontend to a user (versus to an internal users performing a manual sync in the Admin panel), @kieraneglin? Context: Wondering if we need to check look feel of the returned error response in our UI.

Reply:
The raw error message is not something a user would see in our usage of the gem. Rather that would be admin-only.  

Either way, I think this is a reasonable behaviour for the standalone gem as this passes the finch response straight through to the consumer